### PR TITLE
add NewAttributeMapFromMap, mark InitFromMap deprecated

### PR DESCRIPTION
**Pull request by codeboten**
**Description:**
As per issue #3926, adding `NewAttributeMapFromMap` for consistency. As part of this also deprecating the existing `InitFromMap` will will be removed once the contrib repo is updated.

**Link to tracking Issue:** part of #3926 

**Testing:** Updated existing tests

### CODE REVIEW

File: internal/testdata/metric.go [hunk 1]
```diff
@@ -93,8 +93,8 @@ func GenerateMetricsOneCounterOneSummaryMetrics() pdata.Metrics {
 func GenerateMetricsOneMetricNoAttributes() pdata.Metrics {
 	md := GenerateMetricsOneMetric()
 	dps := md.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0).Metrics().At(0).Sum().DataPoints()
-	dps.At(0).Attributes().InitFromMap(map[string]pdata.AttributeValue{})
-	dps.At(1).Attributes().InitFromMap(map[string]pdata.AttributeValue{})
+	pdata.NewAttributeMap().CopyTo(dps.At(0).Attributes())
+	pdata.NewAttributeMap().CopyTo(dps.At(1).Attributes())
```
Comment (bogdandrutu) [inline internal/testdata/metric.go:97]: just call `Clear`

File: internal/testdata/common.go [hunk 2]
```diff
@@ -38,41 +38,41 @@ const (
 )
 
 func initResourceAttributes1(dest pdata.AttributeMap) {
-	dest.InitFromMap(resourceAttributes1)
+	pdata.NewAttributeMapFromMap(resourceAttributes1).CopyTo(dest)
```
Comment (bogdandrutu) [inline internal/testdata/common.go:41]: Probably we can make the resourceAttributes1 to be AttributesMap, separate PR I am ok, but just an issue?

File: internal/testdata/common.go [hunk 3]
```diff
@@ -38,41 +38,41 @@ const (
...
 }
 
 func initSpanLinkAttributes(dest pdata.AttributeMap) {
-	dest.InitFromMap(spanLinkAttributes)
+	pdata.NewAttributeMapFromMap(spanLinkAttributes).CopyTo(dest)
 }
 
 func initMetricAttachment(dest pdata.AttributeMap) {
-	dest.InitFromMap(map[string]pdata.AttributeValue{TestAttachmentKey: pdata.NewAttributeValueString(TestAttachmentValue)})
+	pdata.NewAttributeMapFromMap(map[string]pdata.AttributeValue{TestAttachmentKey: pdata.NewAttributeValueString(TestAttachmentValue)}).CopyTo(dest)
```
Comment (bogdandrutu) [inline internal/testdata/common.go:61]: A simple "UpsertString"?

File: internal/testdata/common.go [hunk 4]
```diff
@@ -38,41 +38,41 @@ const (
...
 }
 
 func initMetricAttributes1(dest pdata.AttributeMap) {
-	dest.InitFromMap(map[string]pdata.AttributeValue{TestLabelKey1: pdata.NewAttributeValueString(TestLabelValue1)})
+	pdata.NewAttributeMapFromMap(map[string]pdata.AttributeValue{TestLabelKey1: pdata.NewAttributeValueString(TestLabelValue1)}).CopyTo(dest)
```
Comment (bogdandrutu) [inline internal/testdata/common.go:65]: A simple "UpsertString"?

File: internal/testdata/common.go [hunk 5]
```diff
@@ -38,41 +38,41 @@ const (
...
 }
 
 func initMetricAttributes12(dest pdata.AttributeMap) {
-	dest.InitFromMap(map[string]pdata.AttributeValue{TestLabelKey1: pdata.NewAttributeValueString(TestLabelValue1), TestLabelKey2: pdata.NewAttributeValueString(TestLabelValue2)}).Sort()
+	pdata.NewAttributeMapFromMap(map[string]pdata.AttributeValue{TestLabelKey1: pdata.NewAttributeValueString(TestLabelValue1), TestLabelKey2: pdata.NewAttributeValueString(TestLabelValue2)}).Sort().CopyTo(dest)
```
Comment (bogdandrutu) [inline internal/testdata/common.go:69]: Two simple "UpsertString"?

File: internal/testdata/common.go [hunk 6]
```diff
@@ -38,41 +38,41 @@ const (
...
 }
 
 func initMetricAttributes13(dest pdata.AttributeMap) {
-	dest.InitFromMap(map[string]pdata.AttributeValue{TestLabelKey1: pdata.NewAttributeValueString(TestLabelValue1), TestLabelKey3: pdata.NewAttributeValueString(TestLabelValue3)}).Sort()
+	pdata.NewAttributeMapFromMap(map[string]pdata.AttributeValue{TestLabelKey1: pdata.NewAttributeValueString(TestLabelValue1), TestLabelKey3: pdata.NewAttributeValueString(TestLabelValue3)}).Sort().CopyTo(dest)
```
Comment (bogdandrutu) [inline internal/testdata/common.go:73]: Two simple "UpsertString"?

File: internal/testdata/common.go [hunk 7]
```diff
@@ -38,41 +38,41 @@ const (
...
 }
 
 func initMetricAttributes2(dest pdata.AttributeMap) {
-	dest.InitFromMap(map[string]pdata.AttributeValue{TestLabelKey2: pdata.NewAttributeValueString(TestLabelValue2)})
+	pdata.NewAttributeMapFromMap(map[string]pdata.AttributeValue{TestLabelKey2: pdata.NewAttributeValueString(TestLabelValue2)}).CopyTo(dest)
```
Comment (bogdandrutu) [inline internal/testdata/common.go:77]: A simple "UpsertString"?

**Review by bogdandrutu — APPROVED** [review]
Let's merge this before the release, and deal with the comments in a follow-up

File: internal/testdata/metric.go (hunk 1 above)
Comment (codeboten) [reply internal/testdata/metric.go:97]: thanks,  this looks much simpler

**Pull request merged**